L1Trigger/Phase2L1GT/plugins/L1GTDeltaCut.h

```diff
@@ -0,0 +1,268 @@
+                     : true;
+        }
+
+        if (inv_mass_checks_) {
+          double precInvMass =
+              scales_.pT_lsb() *
+              std::sqrt(2 * obj1.hwPT().to_double() * obj2.hwPT().to_double() *
+                        (std::cosh(dEta.value() * scales_.eta_lsb()) - std::cos(dPhi.value() * scales_.phi_lsb())));
+
+          double lutInvMass = scales_.pT_lsb() * std::sqrt(2 * static_cast<double>(invMassSqrDiv2) /
+                                                           (dEta < DETA_LUT_SPLIT ? coshEtaLUT_.output_scale()
+                                                                                  : coshEtaLUT2_.output_scale()));
+
+          double error = std::abs(precInvMass - lutInvMass);
+          massErrors.emplace_back(InvariantMassError{error, error / precInvMass, precInvMass});
+        }
+      }
+
+      if (minPTSquared_ || maxPTSquared_) {
+        int64_t pTSquared = obj1.hwPT().to_int64() * obj1.hwPT().to_int64() *
+                                static_cast<int64_t>(std::round(cosPhiLUT_.output_scale())) +
+                            obj2.hwPT().to_int64() * obj2.hwPT().to_int64() *
+                                static_cast<int64_t>(std::round(cosPhiLUT_.output_scale())) +
+                            2 * obj1.hwPT().to_int64() * obj2.hwPT().to_int64() * lutCosDPhi;
+        res &= minPTSquared_ ? pTSquared > std::round(minPTSquared_.value() * cosPhiLUT_.output_scale()) : true;
+        res &= maxPTSquared_ ? pTSquared < std::round(maxPTSquared_.value() * cosPhiLUT_.output_scale()) : true;
+      }
+
+      if (minTransMassSqrDiv2_ || maxTransMassSqrDiv2_) {
+        int64_t transMassDiv2 = obj1.hwPT().to_int64() * obj2.hwPT().to_int64() *
+                                (static_cast<int64_t>(coshEtaLUT_.output_scale()) - lutCosDPhi);
+        res &= minTransMassSqrDiv2_
+                   ? transMassDiv2 > std::round(minTransMassSqrDiv2_.value() * coshEtaLUT_.output_scale())
+                   : true;
+        res &= maxTransMassSqrDiv2_
+                   ? transMassDiv2 < std::round(maxTransMassSqrDiv2_.value() * coshEtaLUT_.output_scale())
+                   : true;
+      }
+
+      return res;
+    }
+
+    static void fillLUTDescriptions(edm::ParameterSetDescription& desc) {
+      edm::ParameterSetDescription coshLUTDesc;
+      L1GTSingleInOutLUT::fillLUTDescriptions(coshLUTDesc);
+      desc.add<edm::ParameterSetDescription>("cosh_eta_lut", coshLUTDesc);
+
+      edm::ParameterSetDescription coshLUT2Desc;
+      L1GTSingleInOutLUT::fillLUTDescriptions(coshLUT2Desc);
+      desc.add<edm::ParameterSetDescription>("cosh_eta_lut2", coshLUT2Desc);
+
+      edm::ParameterSetDescription cosLUTDesc;
+      L1GTSingleInOutLUT::fillLUTDescriptions(cosLUTDesc);
+      desc.add<edm::ParameterSetDescription>("cos_phi_lut", cosLUTDesc);
+    }
+
+    static void fillDescriptions(edm::ParameterSetDescription& desc) {
```

Comment:
An existing (but not enforced) naming convention would be
```suggestion
    static void fillPSetDescription(edm::ParameterSetDescription& desc) {
```
to differentiate more clearly from `fillDescriptions(edm::ConfigurationDescriptions&)`.